server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -588,6 +594,33 @@ private static void parseNonDynamicArray(DocumentParserContext context, final St
                 parseValue(context, lastFieldName);
             }
         }
+        postProcessDynamicArrayMapping(context, lastFieldName);
+    }
+
+    /**
+     * Arrays that have been classified as floats and meet specific criteria are re-mapped to dense_vector.
+     */
+    private static void postProcessDynamicArrayMapping(DocumentParserContext context, String fieldName) {
+        if (context.indexSettings().getIndexVersionCreated().onOrAfter(DYNAMICALLY_MAP_DENSE_VECTORS_INDEX_VERSION)) {
+            final MapperBuilderContext builderContext = context.createDynamicMapperBuilderContext();
+            final String fullFieldName = builderContext.buildFullName(fieldName);
+            final List<Mapper> mappers = context.getDynamicMappers().get(fullFieldName);
```

Comment:
This should use the new `context.getDynamicMappers(fullFieldName);`

Reply:
Thanks for the catch!